cdap-ui/app/cdap/components/PipelineDetails/PipelineRuntimeArgsDropdownBtn/RuntimeArgsModeless/index.js

```diff
@@ -63,22 +63,26 @@ class RuntimeArgsModeless extends PureComponent {
     });
   };
 
-  saveRuntimeArgs = () => {
+  saveRuntimeArgs = async (e) => {
```

Comment:
How does making this function async fix this issue? This is just a click handler right? As far as I debugged this issue, the store is incorrectly updated with old values when the user saves and closes the modeless. 

This is a timing issue which I believe is happening every time we close the modeless where we update the preferences using a PUT call but before that we get the runtime arguments from the backend and update the store. So next time we show the modeless it is still the old value. 

I could have missed something but can you add comment as to why this needs to be async and how that fixes the issue?

Reply:
The timing issue was because `updatePreferences` was asynchronus. The modal was closed and  it was opened before the changes were saved in the backend. So, old arguments were available the next time tests were opening the modal. Making `updatePreferences` synchronous would make sure that the new arguments are propagated through the backend and available to the modeless when user opens it.

Reply:
this is not necessary. Why do you need to convert updatePreference into a promise? it is an observable

Reply:
fixed.